firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/Personalization.java

```diff
@@ -58,9 +76,29 @@ public void logArmActive(@NonNull String key, @NonNull ConfigContainer configCon
       return;
     }
 
-    Bundle params = new Bundle();
-    params.putString(ARM_KEY, metadata.optString(PERSONALIZATION_ID));
-    params.putString(ARM_VALUE, values.optString(key));
-    analyticsConnector.logEvent(ANALYTICS_ORIGIN_PERSONALIZATION, ANALYTICS_PULL_EVENT, params);
+    String choiceId = metadata.optString(CHOICE_ID);
+    if (choiceId.isEmpty()) {
+      return;
+    }
+
+    synchronized (armsCache) {
+      if (choiceId.equals(armsCache.get(key))) {
+        return;
+      }
+      armsCache.put(key, choiceId);
+    }
+
+    Bundle logParams = new Bundle();
+    logParams.putString(ARM_KEY, key);
+    logParams.putString(ARM_VALUE, values.optString(key));
+    logParams.putString(PERSONALIZATION_ID_LOG_KEY, metadata.optString(PERSONALIZATION_ID));
+    logParams.putInt(ARM_INDEX_LOG_KEY, metadata.optInt(ARM_INDEX, -1));
+    logParams.putString(GROUP, metadata.optString(GROUP));
```

Comment:
I'm not sure where the metadata come from? is this logging server side's decision?

Reply:
Yeah, from the server.